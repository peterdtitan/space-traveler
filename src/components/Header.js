import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/images/planet.png';

export default function Header() {
  return (
    <nav className="w-full flex bg-black/80 justify-center md:justify-between items-center py-4 px-6">
      <div className="flex gap-2 items-center">
        <img src={logo} alt="logo" className="md:w-12 w-8 invert" />
        <h1 className="text-sm md:text-3xl text-white">Space Travelers&apos; Hub</h1>
      </div>

      <div className="md:flex justify-between hidden">
        <div className="flex gap-4 p-2 border-r-[2px] border-white">
          <NavLink
            to="/"
            className={({ isActive }) => (isActive ? 'text-yellow-500 p-2 rounded-md underline underline-offset-4' : 'text-white p-2 rounded-md')}
          >
            Rockets
          </NavLink>
          <NavLink
            to="/missions"
            className={({ isActive }) => (isActive ? 'text-yellow-500 p-2 rounded-md underline underline-offset-4' : 'text-white p-2 rounded-md')}
          >
            Missions
          </NavLink>
        </div>
        <div className="p-2 hidden md:flex">
          <NavLink
            to="/profile"
            className={({ isActive }) => (isActive ? 'text-yellow-500 p-2 rounded-md underline underline-offset-4' : 'text-white p-2 rounded-md')}
          >
            My Profile
          </NavLink>
        </div>
      </div>
    </nav>
  );
}
