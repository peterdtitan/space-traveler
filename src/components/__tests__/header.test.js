import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header';

describe('Header component', () => {
  test('renders Header component correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
