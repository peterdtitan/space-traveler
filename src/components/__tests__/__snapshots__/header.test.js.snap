// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header component correctly 1`] = `
<nav
  class="w-full flex bg-black/80 justify-center md:justify-between items-center py-4 px-6"
>
  <div
    class="flex gap-2 items-center"
  >
    <img
      alt="logo"
      class="md:w-12 w-8 invert"
      src="planet.png"
    />
    <h1
      class="text-sm md:text-3xl text-white"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <div
    class="md:flex justify-between hidden"
  >
    <div
      class="flex gap-4 p-2 border-r-[2px] border-white"
    >
      <a
        aria-current="page"
        class="text-black bg-yellow-500 p-2 rounded-md"
        href="/"
      >
        Rockets
      </a>
      <a
        class="text-white bg-black/80 p-2 rounded-md"
        href="/missions"
      >
        Missions
      </a>
    </div>
    <div
      class="p-2 hidden md:flex"
    >
      <a
        class="text-white bg-black/80 p-2 rounded-md"
        href="/profile"
      >
        My Profile
      </a>
    </div>
  </div>
</nav>
`;
